velox/functions/lib/Re2Functions.cpp

```diff
@@ -496,46 +496,72 @@ bool matchExactPattern(
   return input.size() == 0;
 }
 
+// Matches a kRelaxedFixed pattern forward against ASCII input.
+// Returns a pair:
+//   - first: a bool which tells us whether it matches.
+//   - second: an integer which has different meaning depends on 'first'.
+//     - where is cursor in the input when we finished matching if 'first' is
+//     true.
+//     - the sub pattern index if 'first' is false, -1 if it fails not because
+//     of any sub-pattern(e.g. the input is too short).
 std::pair<bool, int32_t> matchRelaxedFixedForwardAscii(
     StringView input,
     const PatternMetadata& patternMetadata,
-    size_t start) {
-  if (input.size() - start < patternMetadata.length()) {
+    size_t start,
+    size_t startSubPatternIndex,
+    size_t endSubPatternIndex) {
+  const auto lengthOfThePattern =
+      patternMetadata.lengthOf(startSubPatternIndex, endSubPatternIndex);
+  // Compare the length first.
+  if (input.size() - start < lengthOfThePattern) {
     return std::make_pair(false, -1);
   }
 
-  for (const auto& subPattern : patternMetadata.subPatterns()) {
+  auto pattenOffset = patternMetadata.subPatterns()[startSubPatternIndex].start;
+  for (auto i = startSubPatternIndex; i < endSubPatternIndex + 1; ++i) {
+    const auto& subPattern = patternMetadata.subPatterns()[i];
     if (subPattern.kind == SubPatternKind::kLiteralString &&
         std::memcmp(
-            input.data() + start + subPattern.start,
+            input.data() + start + (subPattern.start - pattenOffset),
             patternMetadata.fixedPattern().data() + subPattern.start,
             subPattern.length) != 0) {
-      return std::make_pair(false, -1);
+      return std::make_pair(false, i);
     }
   }
 
-  return std::make_pair(true, start + patternMetadata.length());
+  return std::make_pair(true, start + lengthOfThePattern);
 }
 
+// Matches a kRelaxedFixed pattern forward against Unicode input.
+// Returns a pair:
```

Comment:
This description is pretty complex and is a copy-paste from above. It might be better to simply refer to above function.

```
The return values are the same as in matchRelaxedFixedForwardAscii.
```